tests/serverpod_test_server/test/decode_with_type_test.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+void main() {
+  var protocol = Protocol();
+
+  group('Given a JSON string', () {
```

Comment:
Suggestion: Given should be different for all these tests, please make sure that they are self documenting.